staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -184,46 +169,26 @@ func NewCodecFactory(scheme *runtime.Scheme, mutators ...CodecFactoryOptionsMuta
 }
 
 // newCodecFactory is a helper for testing that allows a different metafactory to be specified.
-func newCodecFactory(scheme *runtime.Scheme, serializers []serializerType) CodecFactory {
+func newCodecFactory(scheme *runtime.Scheme, serializers []runtime.SerializerInfo) CodecFactory {
 	decoders := make([]runtime.Decoder, 0, len(serializers))
 	var accepts []runtime.SerializerInfo
 	alreadyAccepted := make(map[string]struct{})
 
 	var legacySerializer runtime.Serializer
 	for _, d := range serializers {
 		decoders = append(decoders, d.Serializer)
-		for _, mediaType := range d.AcceptContentTypes {
-			if _, ok := alreadyAccepted[mediaType]; ok {
-				continue
-			}
-			alreadyAccepted[mediaType] = struct{}{}
-			info := runtime.SerializerInfo{
-				MediaType:        d.ContentType,
-				EncodesAsText:    d.EncodesAsText,
-				Serializer:       d.Serializer,
-				PrettySerializer: d.PrettySerializer,
-				StrictSerializer: d.StrictSerializer,
-			}
-
-			mediaType, _, err := mime.ParseMediaType(info.MediaType)
-			if err != nil {
-				panic(err)
-			}
-			parts := strings.SplitN(mediaType, "/", 2)
-			info.MediaTypeType = parts[0]
-			info.MediaTypeSubType = parts[1]
-
-			if d.StreamSerializer != nil {
-				info.StreamSerializer = &runtime.StreamSerializerInfo{
-					Serializer:    d.StreamSerializer,
-					EncodesAsText: d.EncodesAsText,
-					Framer:        d.Framer,
-				}
-			}
-			accepts = append(accepts, info)
-			if mediaType == runtime.ContentTypeJSON {
-				legacySerializer = d.Serializer
-			}
+		if _, ok := alreadyAccepted[d.MediaType]; ok {
+			continue
+		}
+		alreadyAccepted[d.MediaType] = struct{}{}
+
+		if d.StreamSerializer != nil {
+			cloned := *d.StreamSerializer
+			d.StreamSerializer = &cloned
+		}
+		accepts = append(accepts, d)
```

Comment:
I'm not sure the StreamSerializer copy in the original is anything more than a consequence of hoisting `serializerType`'s `StreamSerializer`, `EncodesAsText`, and `Framer` into a `StreamSerializerInfo`. I clone the input `*StreamSerializerInfo` since anyone who passed a `WithSerializer` option might have saved a reference to it. I doubt it would ever be a problem in practice and I would not feel bad removing the copy here.

Loop variables are scoped per-iteration since Go 1.22, so it shouldn't be necessary to also copy `d` (it still feels wrong to me because I am so used to the previous Go behavior).

Reply:
>so it shouldn't be necessary to also copy d (it still feels wrong to me because I am so used to the previous Go behavior).

Ok, let's do it anyway for readability.